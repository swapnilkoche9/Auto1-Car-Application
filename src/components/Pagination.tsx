import React from 'react'
import constants from '../constants/SystemConstants'
import { PaginationProps } from '../utils/interface'

class Pagination extends React.Component<PaginationProps, {}> {

  // go to the first page
  firstPage = () => {
    this.props.changePageNumber(1)
    this.props.getPageParams(1)
  }
  // go to the previous page
  previousPage = () => {
    const {
      currentPage
    } = this.props

    if (currentPage - 1 !== 0) {
      this.props.changePageNumber(currentPage - 1)
      this.props.getPageParams(currentPage - 1)
    } else {
      this.props.changePageNumber(1)
    }
  }

  // go to the last page

  lastPage = () => {
    const {
      totalPageCount
    } = this.props

    this.props.changePageNumber(totalPageCount)
    this.props.getPageParams(totalPageCount)
  }

  //  go to next page

  nextPage = () => {
    const {
      totalPageCount,
      currentPage
    } = this.props

    if (currentPage + 1 > totalPageCount) {
      this.props.changePageNumber(totalPageCount)
      this.props.getPageParams(totalPageCount)
    } else {
      this.props.changePageNumber(currentPage + 1)
      this.props.getPageParams(currentPage + 1)
    }
  }
  static defaultProps: { currentPage: number };

  render() {
    const {
      currentPage,
      totalPageCount
    } = this.props

    return (
      <div className='pagination'>
        <a href='#view details' className='paginationText orangeText' onClick={this.firstPage}>{constants.FIRST}</a>
        <a href='#view details' className='paginationText orangeText' onClick={this.previousPage}>{constants.PREVIOUS}</a>
        <a href='#view details' className='paginationText grayText'>{`${constants.PAGE_TEXT} ${currentPage} ${constants.OF_TEXT} ${totalPageCount}`}</a>
        <a href='#view details' className='paginationText orangeText' onClick={this.nextPage}>{constants.NEXT}</a>
        <a href='#view details' className='paginationText orangeText' onClick={this.lastPage}>{constants.LAST}</a>
      </div>
    )
  }
}

Pagination.defaultProps = {
  currentPage: 1
}

export default Pagination
