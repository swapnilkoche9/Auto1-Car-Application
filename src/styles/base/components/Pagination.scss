.pagination {
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-top: 35px;
}
.paginationText {
  text-align: start;
  margin: 0;
  color: $dark-gray;
  font-size: $m-size;
  padding-left: $xl-size;
}
.orangeText{
  color: $orange;
  text-decoration: none;
}
.grayText{
  color: $dark-gray;
  text-decoration: none;
}



