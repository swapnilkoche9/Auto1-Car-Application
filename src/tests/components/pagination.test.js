import React from 'react'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Pagination from '../../components/Pagination'

Enzyme.configure({ adapter: new Adapter() })
const { shallow } = Enzyme
let getPageParams
let wrapper
let changePageNumber
beforeEach(() => {
  getPageParams = jest.fn()
  changePageNumber = jest.fn()
  wrapper = shallow(
    <Pagination
      getPageParams={getPageParams}
      totalPageCount={100}
      currentPage={1}
      changePageNumber={changePageNumber}
    />
  )
})
test('should render Pagination correctly', () => {
  expect(wrapper).toMatchSnapshot()
})

test('should render Pagination with alt data correctly', () => {
  wrapper.setProps({
    totalPageCount: 5
  })
  expect(wrapper).toMatchSnapshot()
})

test('should increase the current page by 1', () => {
  wrapper.setProps({
    currentPage: 1
  })
  const instance = wrapper.instance()
  instance.nextPage()
  expect(changePageNumber).toHaveBeenCalled()
  expect(getPageParams).toHaveBeenCalled()
})

test('should decrease the current page by 1', () => {
  wrapper.setProps({
    currentPage: 10
  })
  const instance = wrapper.instance()
  instance.previousPage()
  expect(changePageNumber).toHaveBeenCalled()
  expect(getPageParams).toHaveBeenCalled()
})

test('should skip to the first page', () => {
  wrapper.setProps({
    currentPage: 1
  })
  const instance = wrapper.instance()
  instance.firstPage()
  expect(changePageNumber).toHaveBeenCalled()
  expect(getPageParams).toHaveBeenCalled()
})

test('should skip to the last page', () => {
  wrapper.setProps({
    currentPage: 100
  })
  const instance = wrapper.instance()
  instance.lastPage()
  expect(changePageNumber).toHaveBeenCalled()
  expect(getPageParams).toHaveBeenCalled()
})
