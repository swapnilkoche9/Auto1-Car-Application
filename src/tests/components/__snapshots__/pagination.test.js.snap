// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Pagination correctly 1`] = `ShallowWrapper {}`;

exports[`should render Pagination with alt data correctly 1`] = `ShallowWrapper {}`;
